### ISSUE TITLE
Please document platform support - looks like HippoMocks is Windows-on-x86/ARM only

### ISSUE THREAD

**Issue by jbrwilkinson** [dascandy/hippomocks#34]
We also use Linux-on-x86/64, AIX-on-Power and Solaris-on-Sparc platforms at our office and, from reading the HippoMocks headers, it appears that we cannot use HippoMocks as the framework leverages Windows functions like VirtualProtect and has hand-crafted assembly.

Please can you document this somewhere.

**Comment by dascandy** [comment]
It works on all platforms - except for the static / free-function mocking which only works on all x86 (+64bit) and ARM platforms. I don't have any Power / Sparc systems to test on, so they are effectively unsupported due to lack of hardware. 

If you only mock virtual functions, that functionality is portable to any GCC/MSVC/Clang targeted system (and probably others as well, but not tested). EDG functionality was also present but hasn't been tested in a while, due to lack of any non-commercial EDG to test it with.

**Comment by rzr** [comment]
yes It would be nice to have other platform support, if anyone want to join you're welcome...

I suggest to start with MIPS support

https://github.com/dascandy/hippomocks/pull/50

Also it could be good to have it integrated into debian as it supports architectures and we could get feedback from user, I could help on that.

Keep in touch

**Comment by gza3we** [comment]
Is MinGW on Windows with GCC compiler also a supported platform?
